add dvm progress information to migmigration

alaypatel07 opened the pull request:
Closes #794

dymurray commented on pkg/apis/migration/v1alpha1/owner_reference.go:
```diff
@@ -0,0 +1,34 @@
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func SetOwnerReference(owner metav1.Object, ownerType runtime.Object, dim metav1.Object) {
```
HUGE +1

dymurray reviewed (commented):
VISIACK

pranavgaikwad commented on pkg/controller/migmigration/dvm.go:
```diff
@@ -116,6 +118,21 @@ func (t *Task) hasDirectVolumeMigrationCompleted(dvm *migapi.DirectVolumeMigrati
+func (t *Task) getDVMProgress(runningPods []*migapi.RunningPod) []string {
+	progress := []string{}
+	for _, pod := range runningPods {
+		p := fmt.Sprintf("rsync client pod %s is running", path.Join(pod.Namespace, pod.Name))
```
```suggestion
		p := fmt.Sprintf("Pod %s/%s: Running", pod.Namespace, pod.Name)
		if pod.LastObservedProgressPercent != "" {
		        p = fmt.Sprintf("Pod %s/%s: Running, %s completed", pod.LastObservedProgressPercent)
		}
		progress = append(progress, p)
```

pranavgaikwad replied:
I don't think we absolutely need to show the transfer rate as of now. The UI doesnt have anything for the transfer rate. 

Apart from these, it would be nice if we could show duration at the end of the string, just like Velero Backup / Restore. But again, not a hard requirement.

pranavgaikwad replied:
What happens when a Pod fails / succeds? This progress info is only for in-progress pods.

alaypatel07 replied:
@pranavgaikwad would it be hard to add something in the UI to show the transfer rate? I guess why can we not have the transfer rate in CLI and ignore it in the UI? Feels weird if we have to remove something from the CR just because of UI

sseago reviewed (approved):
LGTM on the DIM changes.

alaypatel07 commented:
@pranavgaikwad https://github.com/konveyor/mig-controller/pull/808/commits/bc78872ead01daabfe8de29ed699bc812cb497fb addresses some of your concerns, please ack this so we can merge this and handle it in follow on, thanks! 

Since this is near to code freeze, want to start merging small PRs and iterate with follow-ons

djwhatle commented on pkg/apis/migration/v1alpha1/owner_reference.go:
```diff
@@ -0,0 +1,34 @@
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func SetOwnerReference(owner metav1.Object, ownerType runtime.Object, dim metav1.Object) {
```
@alaypatel07 could we name `dim metav1.Object` more generically since this looks like it will work on any metav1.Object?

alaypatel07 replied:
since we already have an owner, do you mind if I generically name this `obj`?

djwhatle replied:
Sure, just something other than DIM

djwhatle commented on pkg/controller/directvolumemigration/rsync.go:
```diff
@@ -693,13 +682,14 @@ func (t *Task) createPVProgressCR() error {
+			err := t.Client.Create(context.TODO(), &dvmp)
 			if k8serror.IsAlreadyExists(err) {
-				t.Log.Info("Rsync client progress CR already exists on destination", "namespace", dvp.Namespace, "name", dvp.Name)
+				t.Log.Info("Rsync client progress CR already exists on destination", "namespace", dvmp.Namespace, "name", dvmp.Name)
```
@alaypatel07 I noticed that this gets logged repeatedly during the migration. Do we really need to log this? What can the user do with this info?

alaypatel07 replied:
@djwhatle this was just convention copied from directvolumemigration controller, if we need this fixed, can we do that as a follow up with both controller at the same time?

djwhatle replied:
That works.

djwhatle commented on pkg/controller/migmigration/dvm.go:
```diff
@@ -8,6 +8,7 @@ import (
 	dvmc "github.com/konveyor/mig-controller/pkg/controller/directvolumemigration"
```
FYI @alaypatel07 typically we have put shared consts like `dvmc.MigrationFailed` in `migapi` (the package we alias as migapi) so that we don't end up with circular references (which will break the build) if two controllers need to interact with each other in some way.

djwhatle reviewed (changes requested):
See comments

pranavgaikwad reviewed (approved):
I approve. Let's get the rest of the changes in follow up PRs.

djwhatle reviewed (approved):
Seems like there's a push on this so I don't want to hold this up, but ask that we make a note of the requested changes if they won't be resolved now.
